src/httpServer/server.ts

```diff
@@ -71,19 +69,51 @@ export class OperonHttpServer {
     });
   }
 
-  static registerDecoratedEndpoints(operon : Operon, router : Router)
-  {
+  /**
+   * Register workflow recovery endpoint.
+   * Receives a list of executor IDs and returns a list of workflowUUIDs.
+   */
+  static registerRecoveryEndpoint(operon: Operon, router: Router) {
+    // Handler function that parses request for recovery.
+    const recoveryHandler = async (koaCtxt: Koa.Context, koaNext: Koa.Next) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      const executorIDs = koaCtxt.request.body as string[];
+      operon.logger.info("Recovering workflows for executors: " + executorIDs.toString());
+      const recoverHandles = await operon.recoverPendingWorkflows(executorIDs);
+      operon.recoveryWorkflowHandles.push(...recoverHandles);
+
+      // Return a list of workflowUUIDs being recovered.
+      koaCtxt.body = await Promise.allSettled(recoverHandles.map((i) => i.getWorkflowUUID())).then((results) =>
+        results.filter((i) => i.status === "fulfilled").map((i) => (i as PromiseFulfilledResult<unknown>).value)
```

Comment:
Is there a way to get `fulfilled` from a native JS? For example, `Promise.Status`or something? 🤯

Reply:
I think this is the best way I've found so far...